From: tytan652 <tytan652@tytanium.xyz>
Date: Sat, 11 Apr 2026 16:55:13 +0200
Subject: notification,screenshot: Ensure that the version set is correct

If the implementation set a version greater than the portal itself, the
portal consumer should not have to deal with it.

Origin: https://github.com/flatpak/xdg-desktop-portal/pull/1966
---
 src/notification.c | 2 +-
 src/screenshot.c   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/notification.c b/src/notification.c
index ae9b5ee..a2f229d 100644
--- a/src/notification.c
+++ b/src/notification.c
@@ -1298,7 +1298,7 @@ notification_new (XdpContext              *context,
   notification->impl_version =
     MAX (xdp_dbus_impl_notification_get_version (notification->impl), 1);
   xdp_dbus_notification_set_version (XDP_DBUS_NOTIFICATION (notification),
-                                     notification->impl_version);
+                                     MIN (notification->impl_version, 2));
 
   g_object_bind_property (G_OBJECT (notification->impl), "supported-options",
                           G_OBJECT (notification), "supported-options",
diff --git a/src/screenshot.c b/src/screenshot.c
index 9f0c05b..2e710fe 100644
--- a/src/screenshot.c
+++ b/src/screenshot.c
@@ -542,7 +542,7 @@ screenshot_new (XdpDbusImplScreenshot *impl,
     MAX (xdp_dbus_impl_screenshot_get_version (screenshot->impl), 2);
 
   xdp_dbus_screenshot_set_version (XDP_DBUS_SCREENSHOT (screenshot),
-                                   screenshot->impl_version);
+                                   MIN (screenshot->impl_version, 2));
 
   return screenshot;
 }
